apps/api-runtime/src/main/java/com/akto/runtime/APICatalogSync.java

```diff
@@ -1383,17 +1376,23 @@ public void syncWithDB(boolean syncImmediately, boolean fetchAllSTI) {
         }
 
         loggerMaker.infoAndAddToDb("adding " + writesForParams.size() + " updates for params", LogDb.RUNTIME);
+        int from = 0;
+        int batch = 10000;
 
         long start = System.currentTimeMillis();
-
         if (writesForParams.size() >0) {
-            
-            BulkWriteResult res = 
-                SingleTypeInfoDao.instance.getMCollection().bulkWrite(
-                    writesForParams
-                );
-
-                loggerMaker.infoAndAddToDb((System.currentTimeMillis() - start) + ": " + res.getInserts().size() + " " +res.getUpserts().size(), LogDb.RUNTIME);
+            do {
+
+                List<WriteModel<SingleTypeInfo>> slicedWrites = writesForParams.subList(from, Math.min(from + batch, writesForParams.size()));
+                from += batch;
+                BulkWriteResult res =
+                        SingleTypeInfoDao.instance.getMCollection().bulkWrite(
+                                slicedWrites,
+                                new BulkWriteOptions().ordered(true).bypassDocumentValidation(false)
```

Comment:
Ordered should be set to false, else even if a single write fails, the entire batch will fail